llvm/lib/SYCLLowerIR/GlobalOffset.cpp

```diff
@@ -317,10 +381,8 @@ class GlobalOffset : public ModulePass {
     return {NewFunc, ImplicitOffset};
   }
 
-  static llvm::DenseMap<Function *, MDNode *> getEntryPointMetadata(Module &M) {
-    auto NvvmMetadata = M.getNamedMetadata("nvvm.annotations");
-    assert(NvvmMetadata && "IR compiled to PTX must have nvvm.annotations");
-
+  static DenseMap<Function *, MDNode *>
+  validateKernels(Module &M, SmallVectorImpl<KernelPayload> &KernelPayloads) {
```

Comment:
Not sure why we need a name change here. 'getEntryPointMetadata' sounded better.

Reply:
It previously did the traversing of metadata nodes to find the kernels (https://github.com/intel/llvm/blob/sycl/llvm/lib/Target/NVPTX/SYCL/GlobalOffset.cpp#L321), it felt like now the most important task of it was to validate that kernel doesn't have uses other that LLVM uses `HasUseOtherThanLLVMUsed`. The rest is just re-shuffling the data structure.